src/cargo/ops/vendor.rs

```diff
@@ -255,7 +255,7 @@ fn sync(
         let name = if source_id.is_default_registry() {
             CRATES_IO_REGISTRY.to_string()
         } else {
-            source_id.url().to_string()
+            source_id.as_url().to_string()
```

Comment:
Using the full URL can result in some pretty verbose output, like:

```
[source."git+file:///Users/eric/Proj/rust/cargo/target/tmp/cit/t0/git?rev=17967f9b2816d8f67e36e8fae89e1cc31b37a9ff#17967f9b2816d8f67e36e8fae89e1cc31b37a9ff"]
```

I'm thinking at least it might be worthwhile to trim the precise data so that the `#` part gets removed. Perhaps something like:

```suggestion
            // Remove `precise` since that makes the source name very long,
            // and isn't needed to disambiguate multiple sources.
            source_id.with_precise(None).as_url().to_string()
```